src/app/modules/main/chat_section/model.nim

```diff
@@ -239,230 +251,237 @@ QtObject:
     let item = self.getItemById(id)
     if (not item.isNil):
       return ("", item.position)
-    else:
-      for it in self.items:
-        let item = it.subItems.getItemById(id)
-        if(not item.isNil):
-          return (it.categoryId, item.position)
     return ("", -1)
 
-  proc getSubItemById*(self: Model, id: string): SubItem =
-    for it in self.items:
-      let item = it.subItems.getItemById(id)
-      if(not item.isNil):
-        return item
-
-  proc setActiveItemSubItem*(self: Model, id: string, subItemId: string) =
+  proc setActiveItem*(self: Model, id: string) =
     for i in 0 ..< self.items.len:
-      self.items[i].setActiveSubItem(subItemId)
-
-      if(self.items[i].active):
+      let isChannelToSetActive = (self.items[i].id == id)
+      if self.items[i].active != isChannelToSetActive:
         let index = self.createIndex(i, 0, nil)
-        self.items[i].BaseItem.active = false
+        # Set active channel to true and others to false
+        self.items[i].active = isChannelToSetActive
         self.dataChanged(index, index, @[ModelRole.Active.int])
 
-      if(self.items[i].id == id):
-        let index = self.createIndex(i, 0, nil)
-        self.items[i].BaseItem.active = true
-        self.dataChanged(index, index, @[ModelRole.Active.int])
-
-  proc getItemOrSubItemByIdAsJson*(self: Model, id: string): JsonNode =
-    for it in self.items:
-      if(it.id == id):
-        return it.toJsonNode()
-
-      var found = false
-      let jsonObj = it.subItems.getItemByIdAsJson(id, found)
-      if(found):
-        return jsonObj
-
-  proc getItemOrSubItemByIdAsBase*(self: Model, id: string): BaseItem =
-    for it in self.items:
-      if(it.id == id):
-        return BaseItem(it)
-      let subItem = it.subItems.getItemById(id)
-      if (not subItem.isNil):
-        return BaseItem(subItem)
-
-  proc muteUnmuteItemOrSubItemById*(self: Model, id: string, mute: bool) =
-    for i in 0 ..< self.items.len:
-      if(self.items[i].id == id):
-        let index = self.createIndex(i, 0, nil)
-        self.items[i].BaseItem.muted = mute
-        self.dataChanged(index, index, @[ModelRole.Muted.int])
-        return
-
-      if self.items[i].subItems.muteUnmuteItemById(id, mute):
-        self.items[i].BaseItem.muted = self.items[i].subItems.isAllMuted()
-        return
+  proc changeMutedOnItemById*(self: Model, id: string, muted: bool) =
+    let index = self.getItemIdxById(id)
+    if index == -1:
+      return
+    if(self.items[index].muted == muted):
+      return
+    self.items[index].muted = muted
+    let modelIndex = self.createIndex(index, 0, nil)
+    self.dataChanged(modelIndex, modelIndex, @[ModelRole.Muted.int])
 
-  proc muteUnmuteItemsOrSubItemsByCategoryId*(self: Model, categoryId: string, mute: bool) =
+  proc changeMutedOnItemByCategoryId*(self: Model, categoryId: string, muted: bool) =
```

Comment:
Yeah, you have to implement that method (`setData`) yourself in your model ;)

Anyways, I think this is out of scope of this PR as it would entail a lot of changes on the QMl side too; maybe later